docs/release-process.md

```diff
@@ -0,0 +1,183 @@
+# Release process
+
+This document outlines the process for releasing a new version of the Mezo chain client.
+
+## Determine the release type and version number
+
+The `mezod` repository follows the semver versioning scheme. The first step
+of every release is to determine its type and the new version number.
+
+Compare the Git diff between the planned version (recent main) and the current version running on
+the upgraded chain. Use the following rules to determine the version number:
+
+- If the diff contains consensus breaking changes, the release is a major version upgrade.
+- If the diff does not contain consensus breaking changes, the release is a minor/patch version upgrade.
+
+Next, determine the optional suffix:
+
+- If the release is a stable release ready to be rolled out on the Mainnet chain, the version number
+  should not contain any suffix.
+- If the release is a candidate release intended to be rolled out on the Testnet chain only,
```

Comment:
Shouldn't we include description about a scenario when we upgrade `mainnet` with a stable release version, but not having a release candidate for that version? Take `v2.0.0` as an example. We didn't cut `v2.0.0-rc0` and now `mainnet` will be ahead of testnet.

Reply:
Right, I guess what I'm getting at is we should have `testnet` and `mainnet` aligned and preferably releasing `testnet` first to validate the changes for some time and then upgrade `mainnet`. I realize that there might be "hot" cases when we go straight for `mainnet`, but feels like this should be rather an exception. 
Having said that, maybe we should add this to the description and also add the same in "validator communication" section saying that they should upgrade `testnet` first (when we have a -rc candidate) and expect to have `mainnet` to follow after some time?